qa/abstracts/src/main/java/io/camunda/zeebe/process/test/qa/abstracts/inspections/AbstractProcessDefinitionInspectionsTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2021 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.camunda.zeebe.process.test.qa.abstracts.inspections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.process.test.api.ZeebeTestEngine;
+import io.camunda.zeebe.process.test.inspections.InspectionUtility;
+import io.camunda.zeebe.process.test.qa.abstracts.util.Utilities;
+import io.camunda.zeebe.process.test.qa.abstracts.util.Utilities.ProcessPackNamedElements;
+import org.junit.jupiter.api.Test;
+
+public abstract class AbstractProcessDefinitionInspectionsTest {
+  @Test
+  void testFindStartEventIdByName() {
+    // given
+    Utilities.deployResource(getClient(), ProcessPackNamedElements.RESOURCE_NAME);
+    Utilities.deployResource(getClient(), ProcessPackNamedElements.RESOURCE_NAME);
```

Comment:
❓ Why are we deploying the process twice?

Reply:
This asserts that the `distinct()` in the stream resolving the id from the name works accordingly.

The assumption is that on deploying a resource twice, you would get 2 results containing the same id for a name which would then become one entry.

Reply:
Should we create an explicit test for this instead?

Reply:
That's not quite how Zeebe works. Zeebe will notice that the resource hasn't changed and not actually deploy a 2nd version of the process. If you want to test this you'll need a second process which contains en element with the same element id.

Reply:
I adjusted this. There are 2 different resources deployed now